power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -1003,6 +1109,53 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         return se_input;
     }
 
+    template <bool sym>
+    std::vector<ShortCircuitInput> prepare_short_circuit_input() {
+        assert(is_topology_up_to_date_ && is_parameter_up_to_date<sym>());
+
+        std::vector<ShortCircuitInput> sc_input(n_math_solvers_);
+        for (auto& solver_input : sc_input) {
+            solver_input.fault_bus_indptr = {0};
```

Comment:
@mgovers  we have a separate function just to sort and generate any `indptr` like structure. Please use that:

https://github.com/PowerGridModel/power-grid-model/blob/feature/short-circuit-main-model/power_grid_model_c/power_grid_model/include/power_grid_model/sparse_mapping.hpp